### ISSUE TITLE
New distribution: maximum

### ISSUE THREAD

**Issue by Vilin97** [JuliaStats/Distributions.jl#1643]
Would `Distributions.jl` be the right place for the maximum of n iid random variables? The cdf and pdf are easy to implement, as well as `rand`, and others listed [here](https://juliastats.org/Distributions.jl/stable/extends/#Univariate-Distribution). 

To be precise, given `n` iid random variables `X_1`, ..., `X_n`, I am talking about the distribution of the random variable `M_n = maximum([X_1,...,X_n])`.

**Comment by ParadaCarleton** [comment]
Yes, this would be great! Feel free to make a pull request implementing this!
